### DISCUSSION TITLE
Calculation of volume fraction of texture components

### DISCUSSION THREAD

**Discussion by AbhishekG88** [mtex-toolbox/mtex#1444] [Ask Anything]
Dear Ralf Hielscher and others, 

I am using MTEX 5.8.1 version. 
At the moment, I am using the following syntax but it is showing an error to calculate the volume fraction of texture components. Can you please give me the necessary scripts for this calculation? 
Thanks a lot! 

Kind regards,
Abhishek 

mtexdata Aluminium
plot(ebsd,ebsd.orientations)
odf = calcDensity(ebsd('Aluminium').orientations)
plotSection(odf,'contourf')
mtexColorMap LaboTeX
mtexColorbar

ss = specimenSymmetry('orthorhombic')    
Goss = orientation.byEuler(0*degree,45*degree,0*degree,CS,ss)
vGoss= volume(odf,Goss,11*degree);
![Screenshot mtex ](https://user-images.githubusercontent.com/106551980/171040871-ae925157-04e1-4afb-bd53-01b021c25196.jpg)

**Comment by kilir** [comment]
Hi,
you should skip the specimenSymmetry and should be fine.
Cheers,
Rüdiger

**Comment by AbhishekG88** [comment]
Hi Rüdiger, 

Thank you for your reply. The problem has not been solved after skipping the specimen Symmetry option.  Please see the attached figure. 

![Screenshot mtex ](https://user-images.githubusercontent.com/106551980/171252629-25a61c89-ba35-4a0c-9e89-e2c91941845c.jpg)

**Reply by kilir** [reply]
Oh, I see.
You should use `CS=odf.CS`  or the proper CS e.g. CS{2} or whichever index corresponds to the symmetry of the orientations you used to compute the odf.
Cheers,
Rüdiger

**Comment by AbhishekG88** [comment]
Hi Rüdiger,
Thank you for your quick reply. 

I am using the following -

![Screenshot mtex ](https://user-images.githubusercontent.com/106551980/171273251-98244bcd-b84f-4dec-946c-219f8d7af696.jpg)

And I got the same problem after using ebsd('alu') 

![Screenshot mtex 1](https://user-images.githubusercontent.com/106551980/171273643-4d435187-af06-4aef-8bec-f5cf454c75de.jpg)

In my opinion, maybe the following syntax is not correct 


vGoss= volume(odf,Goss,11*degree);


Matlab is saying that -

Error using ODF/volume (line 31)
Crystal cymmmetry of odf and center needs to be the same!


Thanking you, 
Abhishek

**Reply by kilir** [reply]
Hi Abhishek, please my updated comment above.
Rüdiger

**Comment by AbhishekG88** [comment]
Hi Rüdiger,

Many many thanks. 
Now it is working correctly with  CS=odf.CS 
But I can see some errors (red box in attached photo)


So the final syntax is--

mtexdata Aluminium
plot(ebsd,ebsd.orientations)
odf = calcDensity(ebsd('Alu').orientations)
plotSection(odf,'contourf')
mtexColorMap LaboTeX
mtexColorbar
hold on
plotSection(odf,'contour','lineColor','k','linewidth',2)

CS=odf.CS
Goss = orientation.byEuler(0*degree,45*degree,0*degree,CS)
vGoss= volume(odf,Goss,15*degree)*100


![Screenshot MTEX ](https://user-images.githubusercontent.com/106551980/171470348-63d9b0cc-131e-4f1a-9484-0cda8cba4e9b.jpg)
;
